Mesh_3/include/CGAL/Mesh_3/Construct_initial_points_labeled_image.h

```diff
@@ -0,0 +1,325 @@
+    double y = double(j) * vy + ty;
+    double z = double(k) * vz + tz;
+
+    if (i == 0)              x += 1. / 6. * vx;
+    else if (i == xdim - 1)  x -= 1. / 6. * vx;
+    if (j == 0)              y += 1. / 6. * vy;
+    else if (j == ydim - 1)  y -= 1. / 6. * vy;
+    if (k == 0)              z += 1. / 6. * vz;
+    else if (k == zdim - 1)  z -= 1. / 6. * vz;
+
+    return Point(x, y, z);
+  }
+};
+
+} // end namespace internal
+} // end namespace Mesh_3
+
+/*!
+ * \ingroup PkgMesh3Initializers
+ *
+ * Functor for generating initial points in labeled images.
+ * This functor is a model of the `InitialPointsGenerator` concept,
+ * and can be passed as a parameter to `CGAL::make_mesh_3` using the
+ * `CGAL::parameters::initial_points_generator()` parameter function.
+ *
+ * On images that contain multiple non-connected components,
+ * this functor scans the complete image and
+ * outputs points to initialize each component.
+ *
+ * @tparam C3t3 model of `MeshComplex_3InTriangulation_3`
+ * @tparam MeshDomain model of `MeshDomain_3`
+ *
+ * \sa `CGAL::parameters::initial_points_generator()`
+ * \sa `CGAL::make_mesh_3()`
+ * \sa `CGAL::Construct_initial_points_gray_image`
+ */
+template<typename C3t3, typename MeshDomain>
+struct Construct_initial_points_labeled_image
+{
+  const CGAL::Image_3& image_;
+  const MeshDomain& domain_;
+
+  Construct_initial_points_labeled_image(const CGAL::Image_3& image,
+                                         const MeshDomain& domain)
+    : image_(image)
+    , domain_(domain)
+  { }
+
+  /*!
+  * \brief Constructs at least `n` initial points,
+  * by scanning the image and
+  * collecting points in each object in the image,
+  * even if they are non-connected components.
+  * This ensures that each component will be initialized.
+  *
+  * @tparam OutputIterator model of `OutputIterator` that contains points of type
+  * @todo describe type
+  */
+  template <typename OutputIterator>
+  OutputIterator operator()(OutputIterator pts, int n = 20) const
+  {
+    CGAL_IMAGE_IO_CASE(image_.image(), operator()(pts, CGAL::Identity<Word>(), n));
+    return pts;
+  }
+
+  /*!
+   * \brief Same as above, but a `TransformOperator` that transforms values of the image is used.
+   *
+   * @tparam OutputIterator model of `OutputIterator` that contains points of type
+   *   `MeshDomain::Intersection`
+   * @tparam MeshDomain model of `MeshDomain_3`
+   * @tparam TransformOperator functor that transforms values of the image.
+   *   It must provide the following type:<br>
+   *   `result_type` : a type that supports the '==' operator<br>
+   *   and the following operator:<br>
+   *   `result_type operator()(Word v)`
+   *   with `Word` the type of the image values.
+   * @tparam C3t3 model of `MeshComplex_3InTriangulation_3`
+   */
+  template <typename OutputIterator, typename TransformOperator>
+  OutputIterator operator()(OutputIterator pts, TransformOperator transform, int n = 20) const
```

Comment:
so, it's not a model of `InitialPointsGenerator` and needs to be used like generator --> put into a container --> pass the container via the other parameter ?

Reply:
what about not documenting this operator?
I don't think it is needed in more than internal code

Reply:
this in an implementation detail, let's not document it